src/audio/eq_iir/eq_iir.c

```diff
@@ -587,85 +593,76 @@ static int eq_iir_setup(struct comp_data *cd, int nch)
 /*
  * End of EQ setup code. Next the standard component methods.
  */
-
-static struct comp_dev *eq_iir_new(const struct comp_driver *drv,
-				   const struct comp_ipc_config *config,
-				   const void *spec)
+static int eq_iir_init(struct processing_module *mod)
 {
-	struct comp_dev *dev = NULL;
-	struct comp_data *cd = NULL;
-	const struct ipc_config_process *ipc_iir = spec;
-	size_t bs = ipc_iir->size;
-	int i;
-	int ret;
+	struct module_data *md = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *cfg = &md->cfg;
+	struct comp_data *cd;
+	size_t bs = cfg->size;
+	int i, ret;
 
-	comp_cl_info(&comp_eq_iir, "eq_iir_new()");
+	comp_info(dev, "eq_iir_init()");
 
-	/* Check first before proceeding with dev and cd that coefficients
-	 * blob size is sane.
-	 */
+	/* Check first before proceeding with dev and cd that coefficients blob size is sane */
 	if (bs > SOF_EQ_IIR_MAX_SIZE) {
-		comp_cl_err(&comp_eq_iir, "eq_iir_new(), coefficients blob size %u exceeds maximum",
-			    bs);
-		return NULL;
+		comp_err(dev, "eq_iir_init(), coefficients blob size %u exceeds maximum", bs);
+		return -EINVAL;
 	}
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-	dev->ipc_config = *config;
-
 	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
 	if (!cd)
-		goto fail;
-
-	comp_set_drvdata(dev, cd);
+		return -ENOMEM;
 
+	md->private = cd;
 	cd->eq_iir_func = NULL;
 	cd->iir_delay = NULL;
 	cd->iir_delay_size = 0;
```

Comment:
three lines above are superfluous after an `rzalloc()`, not introduced by this patch though

Reply:
Yep, saving code size is good.